elytron/src/main/java/org/wildfly/extension/elytron/HttpAuthenticationFactoryAddHandler.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.extension.elytron;
+
+import static org.wildfly.common.Assert.checkNotNullParam;
+import static org.wildfly.extension.elytron.ElytronDefinition.commonDependencies;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.concurrent.Callable;
+
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.ProcessType;
+import org.jboss.as.controller.RunningMode;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceBuilder;
+import org.jboss.msc.service.ServiceController;
+import org.wildfly.security.auth.server.HttpAuthenticationFactory;
+
+
+/**
+ * A {@link OperationStepHandler} for adding a {@link HttpAuthenticationFactoryService} for an HttpAuthenticationFactory.
+ *
+ * @author <a href="mailto:aabdelsa@redhat.com">Ashley Abdel-Sayed</a>
+ */
+
+abstract class HttpAuthenticationFactoryAddHandler<T> extends BaseAddHandler {
+
+    private final RuntimeCapability<?> runtimeCapability;
+    private final ServiceController.Mode initialMode;
+    private final ServiceController.Mode embeddedInitialMode;
+
+    HttpAuthenticationFactoryAddHandler(Class<T> serviceType, AttributeDefinition[] attributes, RuntimeCapability<?> runtimeCapability) {
+        this(serviceType, ServiceController.Mode.ACTIVE, attributes, runtimeCapability);
+    }
```

Comment:
The only constructor needed is whatever is invoked by AuthenticationFactoryDefinitions.